From 80ffbcaafc54ad1d4f653d04e35fc982528d6e28 Mon Sep 17 00:00:00 2001
From: anitha3x <anithax.h.chandrasekar@intel.com>
Date: Mon, 4 Feb 2019 16:03:20 +0530
Subject: [PATCH 2/2] Adding only one pbap profile client or server to the list

Issue : - We see that even when the pbab client profile
was present in the list, the pbab server profile was
added again in device details list. Therefore
contact sharing option for two pbap profiles server
and client appeared on UI.

Fix: - Keeping a check to not to add the pbap server
profile if pbab client was present in the device details
list.

Change-Id: Ifa96d9d454ec3ce8502b83cf7d74484ff31e07da
Tracked-On: OAM-75968
Signed-off-by: Gaganpreet kaur <gaganpreetx.kaur@intel.com>
---
 .../settings/bluetooth/BluetoothDetailsProfilesController.java | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/src/com/android/settings/bluetooth/BluetoothDetailsProfilesController.java b/src/com/android/settings/bluetooth/BluetoothDetailsProfilesController.java
index b0ed056..e7a12b3 100644
--- a/src/com/android/settings/bluetooth/BluetoothDetailsProfilesController.java
+++ b/src/com/android/settings/bluetooth/BluetoothDetailsProfilesController.java
@@ -54,6 +54,7 @@ public class BluetoothDetailsProfilesController extends BluetoothDetailsControll
     private LocalBluetoothProfileManager mProfileManager;
     private CachedBluetoothDevice mCachedDevice;
     private PreferenceCategory mProfilesContainer;
+    private boolean mPbapClientAdded = false;
 
     public BluetoothDetailsProfilesController(Context context, PreferenceFragment fragment,
             LocalBluetoothManager manager, CachedBluetoothDevice device, Lifecycle lifecycle) {
@@ -194,9 +195,16 @@ public class BluetoothDetailsProfilesController extends BluetoothDetailsControll
     private List<LocalBluetoothProfile> getProfiles() {
         List<LocalBluetoothProfile> result = mCachedDevice.getConnectableProfiles();
 
+        for (LocalBluetoothProfile profile : result) {
+            if (profile.getProfileId() ==  BluetoothProfile.PBAP_CLIENT) {
+                mPbapClientAdded = true;
+                break;
+            }
+        }
+
         final int pbapPermission = mCachedDevice.getPhonebookPermissionChoice();
         // Only provide PBAP cabability if the client device has requested PBAP.
-        if (pbapPermission != CachedBluetoothDevice.ACCESS_UNKNOWN) {
+        if  (!mPbapClientAdded && (pbapPermission != CachedBluetoothDevice.ACCESS_UNKNOWN)) {
             final PbapServerProfile psp = mManager.getProfileManager().getPbapProfile();
             result.add(psp);
         }
-- 
2.7.4

